chain/client/src/client.rs

```diff
@@ -2374,12 +2374,11 @@ impl Client {
                 state_sync_info.shards.iter().map(|tuple| tuple.0).collect();
             // Notify each shard to sync.
             if notify_state_sync {
-                let shard_layout = self
-                    .epoch_manager
-                    .get_shard_layout(&epoch_id)
-                    .expect("Cannot get shard layout");
-                for &shard_id in &tracking_shards {
-                    let shard_uid = ShardUId::from_shard_id_and_layout(shard_id, &shard_layout);
```

Comment:
What's wrong with the current approach of getting the shard_uid? `shard_ids_to_uids` doesn't really seem like it belongs to the epoch manager.

For all future code, we would really want to move towards only ever using `shard_uid`s and never `shard_id`s but I understand it's not possible here.

Reply:
We already have `shard_id_to_uid` in epoch manager and I needed to convert a vector of shard IDs to shard UIDs in two places. Also, `shard_id_to_uid` calls `get_shard_layout(epoch_id)` so I thought it would be more efficient to call it only once when I have to convert a vector of shard IDs.